internal/replacer/replacer_test.go

```diff
@@ -187,27 +187,81 @@ func TestReplaceWordAllCaps(t *testing.T) {
   }
 }
 
-func TestReplaceWordWithSuffix(t *testing.T) {
+func TestReplaceWordWithPrefixes(t *testing.T) {
+  t.Run("maintain prefix", func(t *testing.T) {
+    var wordmap wordmap.WordMap
+    wordmap.AddOne("-ize", "-ise")
+
+    source := "They Realize that they should not idealize."
+    result := ReplaceAll(source, wordmap)
+
+    expected := "They Realise that they should not idealise."
+    if result != expected {
+      reportIncorrectReplacement(t, expected, result)
+    }
+  })
+  t.Run("omit prefix", func(t *testing.T) {
+    var wordmap wordmap.WordMap
+    wordmap.AddOne("- people", "people")
```

Comment:
This is actually a possibly surprising use-case of the prefix/suffix notation. Consider documenting it and testing it separately (also with suffixes).